pkg/translator/prometheusremotewrite/helper.go

```diff
@@ -540,6 +545,24 @@ func addResourceTargetInfo(resource pcommon.Resource, settings Settings, timesta
 	}
 
 	labels := createAttributes(resource, attributes, settings.ExternalLabels, identifyingAttrs, false, model.MetricNameLabel, name)
+	haveJob := false
+	haveInstance := false
+	for _, l := range labels {
+		if l.Name == model.JobLabel {
+			haveJob = true
+		} else if l.Name == model.InstanceLabel {
+			haveInstance = true
+		}
+		if haveJob && haveInstance {
+			break
+		}
+	}
+
+	if !haveJob && !haveInstance {
+		// We need at least one identifying label to generate target_info.
+		return
+	}
```

Comment:
```suggestion
	haveIdentifier := false
	for _, l := range labels {
		if l.Name == model.JobLabel || l.Name == model.InstanceLabel {
			haveIdentifier = true
			break
		}
	}

	if !haveIdentifier {
		// We need at least one identifying label to generate target_info.
		return
	}
```

We only care that one of them is present, might as well short-circuit as soon as either is found.

Reply:
Good suggestion.